runtime/src/main/cpp/Runtime.cpp

```diff
@@ -30,7 +30,7 @@ struct RuntimeState {
   volatile int executionStatus;
 };
 
-typedef void (*Initializer)(int initialize);
+typedef void (*Initializer)(int initialize, MemoryState* memory);
```

Comment:
Is it actually required to pass `MemoryState` explicitly here?

Reply:
I guess so, at least for deinit case, so decided to make it symmetrical.

Reply:
IIRC, `::memoryState` is (and must be) available during e.g. `Initializer(DEINIT_THREAD_LOCAL_GLOBALS)`.

Question is: what happens if `memory != ::memoryState`? I guess these two must actually be equal. For me this means that `memory` parameter is kinda redundant here. At least until `Initializer` ensures that `memory` and `::memoryState` are consistent.